example/demo/src/inventory/ui.tsx

```diff
@@ -173,142 +175,91 @@ export function getHomeScreen(): ServerDataState {
   }
 }
 
+// tbd: we need a better way to design this API, should ideally come from tamagui package.
+const YStack = createComponentDefinition<typeof t.YStack>('YStack')
+const XStack = createComponentDefinition<typeof t.XStack>('XStack')
+const Image = createComponentDefinition<typeof t.Image>('Image')
+const H2 = createComponentDefinition<typeof t.H2>('H2')
+const SizableText = createComponentDefinition<typeof t.H2>('SizableText')
+const Paragraph = createComponentDefinition<typeof t.H2>('Paragraph')
+const Button = createComponentDefinition<typeof t.H2>('Button')
```

Comment:
I am not sure what is the best way to declare importable components that can be used on the Server, ideally, without having to write this.

I was experimenting with the following:
- import any component on the server (e.g. import { YStack } from "tamagui")
- if "component function" has "validator" property, use it on the client to validate props
- on the client, do "import * as TamaguiComponents from "tamagui"" and pass that to `<Template components={TamaguiComponents} />"

This would allow fully seamless code sharing, with one (possibly deal breaking) caveat - if user minifies code at some point down the road, and "import * as TamaguiComponents from "tamagui"" has different names in the client bundle (e.g. "YStack" is "Y$"), the server would still output "YStack" and fail.

Unfortunately, it would require server code to depend on React / Tamagui, which is not ideal.

Reply:
Can we use a different import?

Client:
```
import { RiseComponents } from "@final-ui/kit"
<Template components={RiseComponents} />
```

```
import { RiseSwitchField } from "@final-ui/kit/components"

// write on the server
<RiseSwitchField />
```

Reply:
Yes, right now, in this PR, we import from the "kit" or "kit/runtime" (on the client). So, your suggestion is pretty much the same, but reverse, which is also fine.

We should however discuss how to provide those wrapper components for the server, not sure what's the best way to do it.

Reply:
Well, the issue is the dependency on `@final-ui/kit` from the server package? Can't we export the actual typed JSX components from the `kit` package?